File: src/abstractFactory/java/factory/DravenFactory.java
package abstractFactory.java.factory;

import abstractFactory.java.draven.DravenAbility;
import abstractFactory.java.draven.DravenConfig;

/**
 * Created by kvirus on 2019/6/7 19:43
 * Email @ caoyingkui@pku.edu.cn
 * <p>
 * |   *******    **      **     **     **
 * |  **            **  **       **  **
 * |  **              **         ***
 * |  **              **         **  **
 * |   *******        **         **     **
 */
public class DravenFactory implements ChampionFactory{
    @Override
    public ChampionAbility getAbilities() {
        return new DravenAbility();
    }

    @Override
    public ChampionConfig getConfig() {
        return new DravenConfig();
    }
}

File: src/abstractFactory/kotlin/factory/DravenFactory.kt
package abstractFactory.kotlin.factory

import abstractFactory.kotlin.draven.DravenConfig
import abstractFactory.kotlin.draven.DravenAbility

/**
 * Created by kvirus on 2019/6/7 20:27
 * Email @ caoyingkui@pku.edu.cn
 *
 * |   *******    **      **     **     **
 * |  **            **  **       **  **
 * |  **              **         ***
 * |  **              **         **  **
 * |   *******        **         **     **
 */
class DravenFactory: ChampionFactory {
    override fun getAbilities(): ChampionAbility {
        return DravenAbility()
    }

    override fun getConfig(): ChampionConfig {
        return DravenConfig()
    }
}